packages/game-passes/contracts/GamePasses.sol

```diff
@@ -1391,14 +1381,16 @@ contract GamePasses is
      *      - Current wallet balance + amount would exceed max per wallet
      * @dev Skips check if maxPerWallet is type(uint256).max (unlimited)
      */
-    function _checkMaxPerWallet(uint256 tokenId, address to, uint256 amount) private view {
+    function _updateAndCheckMaxPerWallet(uint256 tokenId, address to, uint256 amount) private {
         TokenConfig storage config = _tokenStorage().tokenConfigs[tokenId];
 
+        config.mintedPerWallet[to] += amount;
+
         if (config.maxPerWallet == 0) {
             revert ExceedsMaxPerWallet(tokenId, to, amount, 0);
         }
 
-        if (config.maxPerWallet != type(uint256).max && config.mintedPerWallet[to] + amount > config.maxPerWallet) {
+        if (config.maxPerWallet != type(uint256).max && config.mintedPerWallet[to] > config.maxPerWallet) {
             revert ExceedsMaxPerWallet(tokenId, to, amount, config.maxPerWallet);
         }
```

Comment:
Why don't you just check `config.mintedPerWallet[to] + amount > config.maxPerWallet` ?